x-pack/filebeat/input/httpjson/chain.go

```diff
@@ -56,38 +56,60 @@
 
 package httpjson
 
-import (
-	"github.com/elastic/elastic-agent-libs/mapstr"
-	"github.com/elastic/elastic-agent-libs/transport/httpcommon"
-)
-
 // chainConfig for chain request.
 // Following contains basic call structure for each call after normal httpjson
 // call.
 type chainConfig struct {
-	Step stepConfig `config:"step" validate:"required"`
+	Step  *stepConfig  `config:"step,omitempty"`
+	While *whileConfig `config:"while,omitempty"`
 }
 
 // stepConfig will contain basic properties like, request.url,
 // request.method and replace parameter. Each step: request.url
 // will contain replace string with original URL to make a skeleton for the
 // call request.
 type stepConfig struct {
-	Request  requestChainConfig  `config:"request"`
+	Auth     *authConfig         `config:"auth"`
+	Request  requestConfig       `config:"request" validate:"required"`
 	Response responseChainConfig `config:"response,omitempty"`
 	Replace  string              `config:"replace,omitempty"`
 }
 
-type requestChainConfig struct {
-	URL        *urlConfig       `config:"url" validate:"required"`
-	Method     string           `config:"method" validate:"required"`
-	Body       *mapstr.M        `config:"body"`
-	Transforms transformsConfig `config:"transforms"`
-
-	Transport httpcommon.HTTPTransportSettings `config:",inline"`
+// whileConfig will contain basic properties like auth parameters, request parameters,
+// response parameters , a replace parameter and an expression parameter called till.
+// While is similar to stepConfig with the addition of till. Till holds an expression
+// and with the combination of "request.retry.max_attempts" retries a request till the
+// expression is evaluated to "true" or request.retry.max_attempts is exhausted. If
+// request.retry.max_attempts is not specified , the max_attempts is always 1.
+type whileConfig struct {
+	Auth     *authConfig         `config:"auth"`
+	Request  requestConfig       `config:"request" validate:"required"`
+	Response responseChainConfig `config:"response,omitempty"`
+	Replace  string              `config:"replace,omitempty"`
+	Till     *valueTpl           `config:"till" validate:"required"`
```

Comment:
Being a non-native english speaker, I was surprised by the choice of `till` instead of `until`. After checking, now I know both are valid, but originally I though _till_ was an abbreviation only used in casual spoken language.

Given that both mean the same, I think there's reasons to prefer `until` instead of `till`:
- It can confuse non-native speakers (it confused me)
- It can lead to configuration errors (is it `till` or `til`? Or was it `until`?)
- In technical documents and programing languages in general, the use of _until_ is more widespread. [citation needed]

Reply:
I agree with choosing `until`.

Reply:
@adriansr @efd6 , agreed , thanks a lot for the suggestions.. will make the change --  till -> until

Reply:
@adriansr @efd6 made the change to until